File: java/UKL/soal6.java
/*
Soal    : 6 / Soal array 2
A       : 3x2
B       : 2x3
fill    : 3x3
*/

public class soal6 {
  public static void main(String[] args) {   
     int a[][]={
        {1,2},
        {3,4},
        {5,6}
    },
    b[][]={
        {1,2,3},
        {5,6,7}
    },
    h[][]={
        {0,0,0},
        {0,0,0},
        {0,0,0}
    };
    System.out.println("Hasil AxB");

    for(int i=0;i<3;i++){
        for(int j=0;j<3;j++){ 
            for(int k=0;k<2;k++){ 
                h[i][j]=h[i][j]+a[i][k]*b[k][j];
            }
        }
    }
 
    for(int i=0;i<3;i++){
        for(int j=0;j<3;j++){
            System.out.print(h[i][j]+"\t");
        }
    System.out.println();
    }
}
}

File: rust/UKL/soal6.rs
/*
Soal    : 6 / Soal array 2
A       : 3x2
B       : 2x3
fill    : 2x3
*/

fn main() {
    let a=[
        [1,8],
        [3,2],
        [4,6],
    ];
    let b=[
        [9,10,21], //9 + 10 = 21
        [5, 9, 8],
    ];
    let mut fill=[
        [0,0,0],
        [0,0,0],
        [0,0,0]
    ];
    let kolom1=3;
    let kolom2=3;
    for _i in 0..kolom1 {
        for _j in 0..kolom2 {
            for _k in 0..2 {
                fill[_i][_j]=fill[_i][_j]+a[_i][_k]*b[_k][_j];
            }
        }
    }
    for _i in 0..kolom1 {
        for _j in 0..kolom2 {
            print!("{}\t", fill[_i][_j]);
        }
        println!();
    }
}
